sysctl.d/50-coredump.conf.in

```diff
@@ -8,5 +8,8 @@
 # See sysctl.d(5) for the description of the files in this directory,
 # and systemd-coredump(8) and core(5) for the explanation of the
 # setting below.
+#
+# Note: the maximum size of /proc/sys/kernel/core_pattern is
+# currently 128 bytes.
 
-kernel.core_pattern=|@rootlibexecdir@/systemd-coredump %P %u %g %s %t %c %h
+kernel.core_pattern=|@rootlibexecdir@/systemd-coredump _PID=%P _SIGNAL=%s _TIMESTAMP=%t _RLIMIT=%c _HOSTNAME=%h
```

Comment:
please, unless we really really have to, let's just leave the pattern the way it is. last time we changed this it was a compat mess, and this time it would be too. Unless you can bring up really really really strong reasons let's just stick to the pattern the way it is.

Reply:
Our priorities here should be, I guess:

1. keep pattern stable
2. use data from pattern rather than /proc, preferably
3. use /proc where necessary
4. translate pattern parameters matching the namespace when passing off coredump to handler in the container.

Reply:
Please can you explain why you prefer the data from the pattern over the ones in /proc ? That simply makes the number of pattern params we have to translate much bigger.

Reply:
The pattern has been changed because it should more extensible now.

But the main point is that we take this (last) opportunity to change it since even if we would keep the old pattern we still can't support container running any older versions of systemd since we can't figure out which pattern params should be passed on.

Reply:
so, i want that nspawn containers with an old payload work with a new host and vice versa. This means we should use a stable core_pattern, as we can't sensibly configure the core pattern inside the container. i.e. let's stick to the line we already have, and not change it again so that it doesn't matter what runs inside the container as long as it is something vaguely modern.

And if we pass the pattern we should pass it correctly and trust it, in particular as it is quicker and easier to used.

Unless I am mistaken our pattern is roughly stable since bdfd7b2c63cac944a3aa1fc0fafd19f41789e208, i.e. 2016. We dropped an argument from the pattern later again, but that time we didn#t break compat, did we?

Hence, if we can make the payload all the way back to 2016 work, then that's a good thing.

Changing the pattern sucks, and passing incorrect data via the pattern sucks even more, hence we should do the translation when entering the container properly and stick to the pattern we always used, passing properly translated, correct data. And then also use it, as we always did.

Reply:
> so, i want that nspawn containers with an old payload work with a new host and vice versa. 

How can you support the case where a host is running an old version ?? If the host doesn't support invoking the handler in a container then it can't work (whatever the version the container has).

Reply:
> Hence, if we can make the payload all the way back to 2016 work, then that's a good thing.

Last compat breakage I found is f45b8015513d38ee5f7cc361db9c5b88c9aae704 (2018) where an argument (%h) was added before the last one.

But the point is that we can't assume that the handler inside the container uses the last pattern released in 2018. It could be anything older, and supporting all the old released patterns would be pretty ugly.

Reply:
> and passing incorrect data via the pattern sucks even more,

Hmm I don't see why you said that since that's not what this patchset is doing. In fact it drops all arguments from the pattern which need translation (i.e. UID/GID) so we don't have to manually translate them later when invoking the handler inside the container.

Reply:
> How can you support the case where a host is running an old version ?? If the host doesn't support invoking the handler in a container then it can't work (whatever the version the container has).

Well, sure. What I tried to say is that we should stick to an established pattern so that on version changes we don't necessarily break stuff, and in particular we can support new host with semi-old container just like that if we want to, and I think that's worth something.

Reply:
> Last compat breakage I found is [f45b801](https://github.com/systemd/systemd/commit/f45b8015513d38ee5f7cc361db9c5b88c9aae704) (2018) where an argument (%h) was added before the last one.
> 
> But the point is that we can't assume that the handler inside the container uses the last pattern released in 2018. It could be anything older, and supporting all the old released patterns would be pretty ugly.

Hmm, true, I wasn#t aware of this change in 2018... That said, it probably was a change with a good reason — after all the container's hostname is otherwise not trivially acquired from just the PID. i.e. /proc/$PID itself doesn't let us know the hostname, right?

i.e. if we have a good reason to change the pattern then we can change the pattern, but i don't think we should do that unless we have a good reason.

Reply:
When the core dump is processed in the container, the hostname is automatically and already logged by journald. Therefore `COREDUMP_HOSTNAME` is a duplicate of `_HOSTNAME` in this case.